rslib/src/storage/card/mod.rs

```diff
@@ -266,16 +266,23 @@ impl super::SqliteStorage {
     /// returns false or no more cards found.
     pub(crate) fn for_each_new_card_in_active_decks<F>(
         &self,
+        random: bool,
         reverse: bool,
         mut func: F,
     ) -> Result<()>
     where
         F: FnMut(NewCard) -> bool,
     {
         let mut stmt = self.db.prepare_cached(&format!(
-            "{} ORDER BY due {}, ord asc",
+            "{} ORDER BY {}",
             include_str!("active_new_cards.sql"),
-            if reverse { "desc" } else { "asc" }
+            if random {
+                "random()"
```

Comment:
Review card sorting uses a different approach:

            ReviewOrderSubclause::Random => "fnvhash(id, mod)",

This keeps things mostly random, but will give reproducible results when starting from the same state, which can be useful in debugging and the close+reopen cases. It would be nice to have the same behaviour here.

The old random insertion order+due gather order ensured siblings were introduced together in template order, whereas currently this code would show templates in a random order. We'll probably want to offer both options, but that presents us with a bit of a challenge:

- `order by fnvhash(nid), ord` would always show notes in the same order
- `order by fnvhash(nid, mod), ord` would separate the notes, as each of the cards will have a different mtime.
- using the mtime of the note would keep the notes together, but we don't have easy access to it, and notes change infrequently
- `fnvhash(nid, $daysElapsed), ord` would mean the position of a given note changes per day. This would have the disadvantage of showing new cards in the same order again if the user were to study new cards, reset them to new, and study them again. But in the absence of a better solution, it may be our best bet. WDYT?

Reply:
(you might also want to check out the enum use in that review sorting code, as it scales a bit better than separate boolean flags)

Reply:
> fnvhash(nid, $daysElapsed), ord would mean the position of a given note changes per day. This would have the disadvantage of showing new cards in the same order again if the user were to study new cards, reset them to new, and study them again. But in the absence of a better solution, it may be our best bet. WDYT?

Makes sense. This user should use a filtered deck. 🙂